### PR TITLE
Add snippets as a language

### LINKED ISSUE #189176: Add snippets as a language
There's a good community grammar for snippets at https://github.com/jeff-hykin/better-snippet-syntax, which would improve our snippets experience. We could include this as one of our built in languages.

### PR DESCRIPTION
**Pull request by alexr00**
Fixes https://github.com/microsoft/vscode/issues/189176

### CODE REVIEW

**Comment by jeff-hykin** [comment]
> I'm going to get some more feedback from the dev team to see if we want to include this a language since it will always show in the language picker.

Is there an altenative that keeps the syntax highlighting but "snippets" doesn't show up in the language picker?

**Comment by alexr00** [comment]
Feedback was positive, including that it shows in the language picker. I will merge at the beginning of the July iteration for maximum time in Insiders before release.

File: extensions/json/syntaxes/snippets.tmLanguage.json
Comment (alexr00) [inline extensions/json/syntaxes/snippets.tmLanguage.json:1]: @jeff-hykin, just FYI, as part of the script that we'll update the grammar with I've made the scope `json.snippets` (changed from `json.comments`).

Comment (jeff-hykin) [reply extensions/json/syntaxes/snippets.tmLanguage.json:1]: that might break some people's theme settings. It's not that big of deal, but you might want to switch to `json.comments.snippets` which should maintain backwards compatibility

**Comment by gjsjohnmurray** [comment]
Will there also be a "New snippet" snippet? Something like this:
```json
	"New snippet" : {
		"scope": "snippets",
		"body": [
			"\"${1:name}\": {",
				"\t\"scope\": \"$2\",",
				"\t\"description\": \"$3\",",
				"\t\"body\": [",
					"\t\t\"$0\"",
				"\t]",
			"},"
		]
	},
```

**Comment by RedCMD** [comment]
> Will there also be a "New snippet" snippet?

for intellisense?
that could easily be done

**Comment by alexr00** [comment]
@jrieken, FYI about [this ](https://github.com/microsoft/vscode/pull/185017#issuecomment-1594657028)possible "escape for snippet" command.

**Comment by jrieken** [comment]
that's like https://github.com/microsoft/vscode/issues/26706?

**Review by Tyriar — CHANGES_REQUESTED** [review]
@alexr00 can we get a linked issue for this to ensure we verify/test?

**Comment by alexr00** [comment]
Added! https://github.com/microsoft/vscode/issues/189176

**Review by Tyriar — APPROVED** [review]

**Pull request merged**